`[if debug]` ignored by `spawn` and generates C error

### Describe the bug

`spawn`ing a function that has the `[if debug]` attribute should only spawn if `-cg` flag is passed to the compiler. But right now it still attempts to spawn said function even when `-cg` is omitted.

### Reproduction Steps

```v
fn main() {
	spawn print_messages_to_console_while_gg_is_gging()
	// run gg context
}

[if debug]
fn print_messages_to_console_while_gg_is_gging() {
	println('doing something')
}
```

### Expected Behavior

Expected the line with `spawn` keyword to be ignored unless the V file was compiled with the `-cg` flag. 

### Current Behavior

With `v -cg run test.v` it runs normally. But with `v run test.v` it generates this error.
```
==================
C:/Users/imado/AppData/Local/Temp/v_0/test.15143203809510131553.tmp.c:6816: warning: implicit declaration of function 'tcc_backtrace'
C:/Users/imado/AppData/Local/Temp/v_0/test.15143203809510131553.tmp.c:12588: error: 'main__print_messages_to_console_while_gg_is_gging' undeclared
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.1 68cbf27.5d1f87c

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.1 68cbf27.5d1f87c
OS: windows, Microsoft Windows 11 Pro v22621 64-bit
Processor: 16 cpus, 64bit, little endian, 

getwd: C:\frostbyte
vexe: C:\Users\imado\v\v.exe
vexe mtime: 2023-09-13 18:04:23

vroot: OK, value: C:\Users\imado\v
VMODULES: OK, value: C:\Users\imado\.vmodules
VTMP: OK, value: C:\Users\imado\AppData\Local\Temp\v_0

Git version: git version 2.33.1.windows.1
Git vroot status: 0.4.1-66-g5d1f87c5 (4 commit(s) behind V master)
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,
operable program or batch file.

thirdparty/tcc status: thirdparty-windows-amd64 e90c2620
```

--- a/vlib/v/gen/c/spawn_and_go.v
+++ b/vlib/v/gen/c/spawn_and_go.v
@@ -12,6 +12,9 @@ enum SpawnGoMode {
 }
 
 fn (mut g Gen) spawn_and_go_expr(node ast.SpawnExpr, mode SpawnGoMode) {
+	if node.call_expr.should_be_skipped {
+		return
+	}
 	is_spawn := mode == .spawn_
 	is_go := mode == .go_
 	if is_spawn {

--- a/vlib/v/tests/spawn_with_cond_fncall_test.v
+++ b/vlib/v/tests/spawn_with_cond_fncall_test.v
@@ -1,0 +1,8 @@
+fn test_main() {
+	spawn print_messages_to_console_while_gg_is_gging()
+}
+
+@[if debug]
+fn print_messages_to_console_while_gg_is_gging() {
+	println('doing something')
+}

